OpcUaClient/Converter/ModelNodeIdToUaNodeId.cpp

```diff
@@ -14,49 +15,88 @@
 #include <easylogging++.h>
 
 namespace Umati {
-	namespace OpcUa {
-		namespace Converter {
-		enum NodeID_Type 
-			{   NODEID_NUMERIC = (int)'i', 
-				NODEID_STRING = (int)'s', 
-				NODEID_GUID = (int)'g',
-				NODEID_BYTESTRING = (int)'b'
-			};
-
-
-			ModelNodeIdToUaNodeId::ModelNodeIdToUaNodeId(const ModelOpcUa::NodeId_t &modelNodeId,
-														 const std::map<std::string, uint16_t> &uriToID)
-					: ModelToUaConverter(uriToID) {
-
-				int type = (int)(modelNodeId.Id.substr(0,1)[0]);
-				std::string subs = modelNodeId.Id.substr(2);	
-				switch (type)
-				{
-				case NODEID_NUMERIC:
-				    m_nodeId = open62541Cpp::UA_NodeId(getNsIndexFromUri(modelNodeId.Uri),std::stoi(subs));
-					break;
-
-				case NODEID_STRING:					
-					m_nodeId = open62541Cpp::UA_NodeId(getNsIndexFromUri(modelNodeId.Uri),subs);
-					break;
-
-				case NODEID_GUID:
-					LOG(ERROR) << "Conversion to GUID not implemented.";
-					throw "Conversion to GUID not implemented.";
-					break;
-
-				case NODEID_BYTESTRING:
-					LOG(ERROR) << "Conversion to ByteString not implemented.";
-					throw std::invalid_argument("Conversion to ByteString not implemented.");
-					break;
-				
-				default:
-					LOG(ERROR) << "Identifier type not valid";
-					throw "Identifier type not valid";
-					break;
-				}
-
-		}
-	}	
+namespace OpcUa {
+namespace Converter {
+
+
+// inverse method of the hexmap used in open62541
+char hexmapValue(char hex) {
+    if(hex >= '0' && hex <= '9')
+        return hex - '0';
+    else if(hex >= 'a' && hex <= 'f')
+        return hex - 'a' + 10;
+    else if(hex >= 'A' && hex <= 'F')
+        return hex - 'A' + 10;
+    else
+	    LOG(ERROR) << "Invalid char in Guid String";
+        throw std::invalid_argument("Invalid char in Guid String");
+ 
+}
+
+//Convert an hex string to a UA_GUID
+UA_Guid hex_to_UA_Guid(const char* hex) {
```

Comment:
@ccvca I think this method is very generic. Is there an better place for this? e.g. open62541Cpp or the open62541 Lib itself?

Reply:
GUID parsing already part of open62541 https://github.com/open62541/open62541/blob/1f47491d0d9a406630a9de1c1d21eb1987885497/include/open62541/types.h#LL324C3-L324C3

I think the whole parsing (including Bytestring) can be replaces with this function (Added Mar 2020, 6 months after the EMO 2019 ):
https://github.com/open62541/open62541/blob/1f47491d0d9a406630a9de1c1d21eb1987885497/include/open62541/types.h#L423

Reply:
thank you - haven´t seen I add this to the branch

Reply:
fixed